### ISSUE TITLE
[Charli] Habanero returns sessionSigs but timeout on pkpWalletInstance.signMessage("message")

### ISSUE THREAD

**Issue by zach-is-my-name** [LIT-Protocol/Issues-and-Reports#34]
### Description of the issue

I keep getting network errors when requesting sessionSigs with provider.getSessionSigs().

![Screenshot from 2024-05-09 21-09-37](https://uploads.linear.app/d4b33f5f-8fa9-4719-893b-7305458eafa3/96788ae2-8ebb-4f6b-9350-576b9e7b83a8/72949bb1-4996-49c9-b540-7ae3f9c7bded?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Q0YjMzZjVmLThmYTktNDcxOS04OTNiLTczMDU0NThlYWZhMy85Njc4OGFlMi04ZWJiLTRmNmItOTM1MC01NzZiOWU3YjgzYTgvNzI5NDliYjEtNDk5Ni00OWM5LWI1NDAtN2FlM2Y5YzdiZGVkIiwiaWF0IjoxNzE1MzMwODczLCJleHAiOjMzMjg1ODkwODczfQ.m79BmAnET1ID-f1xUhXt0dwddphN7UU-rTwyAbOiDEM)

\*Note: I'm clearing localstorage before every run.
So \*`const [sessionSigs, setSessionSigs] = useLocalStorage("sessionSigs"); shouldn't be a concernt` shouldn't be a concern.

```js
// useLitSession.tsx
import {Wallet} from '@ethersproject/wallet'
import { useCallback, useState } from 'react';
import { AuthMethod, SessionSigs } from '@lit-protocol/types';
import { getProviderByAuthMethod } from '../../utils/lit';
import { LitAbility, LitActionResource } from '@lit-protocol/auth-helpers';
import { IRelayPKP } from '@lit-protocol/types';
import useLocalStorage from '@rehooks/local-storage';
import { litNodeClient } from '@/utils/litClients';
 
export default function useLitSession() {
  const [sessionSigs, setSessionSigs] = useLocalStorage("sessionSigs");
  const [sessionLoading, setLoading] = useState(false);
  const [sessionError, setError] = useState();
 
  const initSession = useCallback(
    async (authMethod: AuthMethod, pkp: IRelayPKP): Promise => {
      setLoading(true);
      setError(undefined);
 
      try {
        console.log("run initSession");
        console.log('authMethod', authMethod)
        console.log('pkp', pkp)
 
        let provider;
        try {
          provider = getProviderByAuthMethod(authMethod);
          if (!provider) {
            throw Error('no provider object');
          }
        } catch (e) {
          console.error('error obtaining provider', e);
        }
 
        if (provider && !sessionSigs) {
          const privateKey = process.env.NEXT_PUBLIC_LIT_CAPACITY_TOKEN_WALLET_DEV as string;
          const walletWithCapacityCredit = new Wallet(privateKey);
 
          const expiration = new Date(Date.now() + 1000 * 60 * 60 * 24 * 7).toISOString(); // 1 week
          const resourceAbilities = [{ resource: new LitActionResource('*'), ability: LitAbility.PKPSigning }];
          const capacityTokenIdStr = process.env.NEXT_PUBLIC_LIT_CAPACITY_TOKEN_ID_STRING_H as string;
 
          const { capacityDelegationAuthSig } = await litNodeClient.createCapacityDelegationAuthSig({
            // uses: '1',
            dAppOwnerWallet: walletWithCapacityCredit,
            capacityTokenId: capacityTokenIdStr,
            delegateeAddresses: [pkp.ethAddress],
          });
 
          const sessionSigs: SessionSigs = await provider.getSessionSigs({
            authMethod,
            pkpPublicKey: pkp.publicKey,
            sessionSigsParams: {
              chain: 'ethereum',
              expiration,
              resourceAbilityRequests: resourceAbilities,
              capacityDelegationAuthSig,
            },
            litNodeClient,
          }).catch(error => { console.error(error); throw new Error('error getSessionSigs') });
 
          console.log(`setting sessionSigs: `, sessionSigs);
          setSessionSigs(sessionSigs);
        }
      } catch (e) {
        const error = e as Error;
        console.error("initSession: error", error);
        setError(error);
      } finally {
        setLoading(false);
      }
    },
    [setSessionSigs]
  );
 
  return {
    sessionSigs,
    initSession,
    sessionLoading,
    sessionError,
  };
}
```

```
{
    "errorKind": "Unexpected",
    "errorCode": "NodeUnknownError",
    "status": 400,
    "message": "Unknown error occured",
    "correlationId": "lit_c8b7f6cd9b4c6",
    "details": [
        "unexpected error: ECDSA signing failed: unexpected error: unexpected error: Could not process presign: unexpected error: Unable to execute presignature: unexpected error: Received abort from channel for protocol presign, txn_prefix: lit_c8b7f6cd9b4c6 at SystemTime { tv_sec: 1715306781, tv_nsec: 847507783 }"
    ]
}
 
{
    "errorKind": "Validation",
    "errorCode": "NodeRpcError",
    "status": 502,
    "message": "Lit nodes failed to complete the RPC call possibly due to RPC servers being down or because the RPC call is making an incorrect smart contract call that reverts",
    "correlationId": "lit_c8b7f6cd9b4c6",
    "details": [
        "Error fetching block from hash",
        "validation error: failed to get block when fetching blocks for query: Deserialization Error: expected value at line 1 column 1. Response: error code: 502"
    ]
}```
```

**Comment by zach-is-my-name** [comment]
Curiously though I do get one OK response: 
![Screenshot from 2024-05-09 21-42-30](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/2ad34629-6367-451e-be02-2e8604d9d664)
![Screenshot from 2024-05-09 21-43-28](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/d66da99e-f396-45dd-b57d-e341914f155a)

The balance of the Bad Responses share the same req body as the OK response:
![Screenshot from 2024-05-09 21-44-06](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/766857c2-6ebb-4a3d-b336-a210ea52d658)

**Comment by zach-is-my-name** [comment]
Updated issue title as another report from SDK issues reports something similar.

**Comment by joshLong145** [comment]
@zach-is-my-name could you try again? this is an internal timeout on the network's rpc connection. In my own test i was able to get back a signed session.

**Comment by zach-is-my-name** [comment]
![Screenshot from 2024-05-10 10-39-55](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/3df6073a-465f-4cfb-abb4-a75adffd91bd)
I'm Chetumal time.  (ie Ramsey -1) 😉

**Comment by zach-is-my-name** [comment]
![Screenshot from 2024-05-10 10-44-19](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/fc7fe546-b841-4d6a-8107-dd982d90d951)

**Comment by glitch003** [comment]
yes, there are different RPC servers for manzano and habanero, and you can pass these into the AuthClient:

```
export const RELAY_URL_HABANERO = 'https://habanero-relayer.getlit.dev';
export const RELAY_URL_MANZANO = 'https://manzano-relayer.getlit.dev';
```

This is a bug - we should use the relayer from the correct network automatically.  @joshLong145 

So, for habanero, you could use 

```
const litAuthClient = new LitAuthClient({litRelayConfig: { relayUrl: "https://habanero-relayer.getlit.dev", relayApiKey: "yourapikeyhere" }})
```

**Comment by zach-is-my-name** [comment]
Thanks @glitch003.  Would I also need new a api key than the one I was issued for cayenne?  If yes, maybe Wyatt's available to connect with me on discord to obtain that.  I'm userzach over there

**Comment by spacesailor24** [comment]
@zach-is-my-name Just confirmed the same API key can be used across all the networks

**Comment by zach-is-my-name** [comment]
Thanks @spacesailor24 !

**Comment by joshLong145** [comment]
@zach-is-my-name 
Just a heads up
if you provide a `litNodeClient` as an option to `LitAuthClient` you do not have to manually pass the `relayUrl`

```js
const litAuthClient = new LitAuthClient({
    litNodeClient: "" // your instance of `LitNodeClient`
    litRelayConfig: { relayUrl:  relayApiKey: "yourapikeyhere" }
});
```

**Comment by zach-is-my-name** [comment]
> @zach-is-my-name Just a heads up if you provide a `litNodeClient` as an option to `LitAuthClient` you do not have to manually pass the `relayUrl`
> 
> ```js
> const litAuthClient = new LitAuthClient({
>     litNodeClient: "" // your instance of `LitNodeClient`
>     litRelayConfig: { relayUrl:  relayApiKey: "yourapikeyhere" }
> });
> ```
> 
> as the Auth Client will pick the RPC URL based on what the instance of `LitNodeClient` is configured for.

Even better.  Thanks Josh

**Comment by zach-is-my-name** [comment]
Wait I think I was doing that:
```
// litClients.ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitAuthClient } from '@lit-protocol/lit-auth-client';
import { LIT_NETWORKS_KEYS } from '@lit-protocol/types';

const litNetwork = process.env.NEXT_PUBLIC_LIT_NETWORK_M as LIT_NETWORKS_KEYS;
if (!litNetwork) throw new Error('problem importing litNetwork env')
export const litNodeClient: LitNodeClient = new LitNodeClient({
  alertWhenUnauthorized: false,
  litNetwork,
  checkNodeAttestation: false,
  debug: true
});

export const litAuthClient: LitAuthClient = new LitAuthClient({
  debug: true,
  litRelayConfig: {
    relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
    relayUrl: process.env.NEXT_PUBLIC_LIT_RELAY_URL_M
  },
  litNodeClient,
});

```

**Comment by zach-is-my-name** [comment]
Just tested this:
provider.getSessionSigs(..) success
export const litNodeClient: LitNodeClient = new LitNodeClient({
  alertWhenUnauthorized: false,
  litNetwork,
  checkNodeAttestation: false,
  debug: true
});

export const litAuthClient: LitAuthClient = new LitAuthClient({
  debug: true,
  litRelayConfig: {
    relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
    relayUrl: process.env.NEXT_PUBLIC_LIT_RELAY_URL_M
  },
  // litNodeClient,
});

console.log("******TYPEOF LIT AUTH CLIENT******", typeof litAuthClient)
console.log("******LIT AUTH CLIENT******", litAuthClient)

But if I include the litNodeClient in the authClient constructor param object it fails with the timeout 

export const litAuthClient: LitAuthClient = new LitAuthClient({
  debug: true,
  litRelayConfig: {
    relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
    relayUrl: process.env.NEXT_PUBLIC_LIT_RELAY_URL_M
  },
  // litNodeClient,
});

**Comment by joshLong145** [comment]
@zach-is-my-name 
Thanks for the additional source code. I was able to find a bug in our `LitAuthClient` constructor which was overriding the `relayUrl` I have fixed this and will be publishing a new version with this included. So once this is published in `5.0.1` you will no longer need to set your own `relayUrl`.

- https://github.com/LIT-Protocol/js-sdk/pull/465

**Comment by zach-is-my-name** [comment]
Thanks Josh I just saw that.  Much appreciated

**Comment by zach-is-my-name** [comment]
@glitch003 @joshLong145 I'm very sorry but after returning the provider.getSessionSigs once, maybe twice on manzano and habenero, I can't get seem to get them any more.  Additionally I don't think think the sig shares were ever valid because my app immediately uses PKPEthers's signMessage to sign a Supabase Auth JWT and that did timeout even when the call to provider.getSessionSigs did not.  
<details>
<summary>My Relevant App Code</summary>
```
// litClients.ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitAuthClient } from '@lit-protocol/lit-auth-client';
import { LIT_NETWORKS_KEYS } from '@lit-protocol/types';

const litNetwork = process.env.NEXT_PUBLIC_LIT_NETWORK_M as LIT_NETWORKS_KEYS;
const relayUrl = process.env.NEXT_PUBLIC_LIT_RELAY_URL_M;

export const litNodeClient: LitNodeClient = new LitNodeClient({
  alertWhenUnauthorized: false,
  litNetwork,
  checkNodeAttestation: false,
  debug: true
});

console.log('litNetwork', litNetwork, typeof litNetwork)
console.log('relayUrl', relayUrl, typeof relayUrl)
export const litAuthClient: LitAuthClient = new LitAuthClient({
  debug: true,
  litRelayConfig: {
    relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
    relayUrl
  },
});
```
```
//calling this now (despite the overhead)
// useLitClients.ts
import {Wallet} from '@ethersproject/wallet'
import { LitAuthClient } from '@lit-protocol/lit-auth-client';
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { useEffect } from 'react';

const useLitClients = (litNodeClient: LitNodeClient, litAuthClient: LitAuthClient) => {
  useEffect(() => {
      const connectClients = async () => {
        await litNodeClient.connect().catch(error => {console.error(error); throw new Error('error litNodeClient.connect error')});
    };
    void (async () => {
      await connectClients().catch(error => {console.error(error); throw new Error('error connectClients()')});
    })();
    return () => {
      void (async () => {
        await litNodeClient.disconnect().catch(error => {console.error(error); throw new Error('error disconnect()')});
      })();
    };
  }, [litNodeClient]);
  return { litNodeClient, litAuthClient };
};

export default useLitClients;
```
```
//Here's where I call provider.getSessionSigs()
// useLitSession.tsx
import {Wallet} from '@ethersproject/wallet'
// import {hexlify} from '@ethersproject/bytes'
import { useCallback, useState } from 'react';
import { AuthMethod, SessionSigs } from '@lit-protocol/types';
import { getProviderByAuthMethod } from '../../utils/lit';
import { LitAbility, LitActionResource } from '@lit-protocol/auth-helpers';
import { IRelayPKP } from '@lit-protocol/types';
import useLocalStorage from '@rehooks/local-storage';
import { litNodeClient } from '@/utils/litClients';

export default function useLitSession() {
  const [sessionSigs, setSessionSigs] = useLocalStorage<SessionSigs>("sessionSigs");
  const [sessionLoading, setLoading] = useState<boolean>(false);
  const [sessionError, setError] = useState<Error>();

  const initSession = useCallback(
    async (authMethod: AuthMethod, pkp: IRelayPKP): Promise<void> => {
      setLoading(true);
      setError(undefined);

      try {
        console.log("run initSession");

        const resourceAbilities = [{ resource: new LitActionResource('*'), ability: LitAbility.PKPSigning }];
        const expiration = new Date(Date.now() + 1000 * 60 * 60 * 24 * 7).toISOString(); // 1 week

        let provider;
        try {
          provider = getProviderByAuthMethod(authMethod);
          if (!provider) {
            throw Error('no provider object');
          }
        } catch (e) {
          console.error('error obtaining provider', e);
        }

        if (provider && !sessionSigs) {
          const privateKey = process.env.NEXT_PUBLIC_LIT_CAPACITY_TOKEN_WALLET_DEV;
          if (!privateKey) throw new Error("problem importing process.env.NEXT_PUBLIC_LIT_CAPACITY_TOKEN_WALLET_DEV;")
          const walletWithCapacityCredit = new Wallet(privateKey);
          const capacityTokenIdStr = process.env.NEXT_PUBLIC_LIT_CAPACITY_TOKEN_ID_STRING_M;

          const { capacityDelegationAuthSig } = await litNodeClient.createCapacityDelegationAuthSig({
            uses: '1',
            dAppOwnerWallet: walletWithCapacityCredit,
            capacityTokenId: capacityTokenIdStr,
            delegateeAddresses: [pkp.ethAddress],
          });

          const sessionSigs: SessionSigs = await provider.getSessionSigs({
            authMethod,
            pkpPublicKey: pkp.publicKey,
            sessionSigsParams: {
              chain: 'ethereum',
              expiration,
              resourceAbilityRequests: resourceAbilities,
              capacityDelegationAuthSig,
            },
            litNodeClient,
          }).catch(error => { console.error(error); throw new Error('error getSessionSigs') });

          console.log(`setting sessionSigs: `, sessionSigs);
          setSessionSigs(sessionSigs);
        }
      } catch (e) {
        const error = e as Error;
        console.error("initSession: error", error);
        setError(error);
      } finally {
        setLoading(false);
      }
    },
    [setSessionSigs]
  );

  return {
    sessionSigs,
    initSession,
    sessionLoading,
    sessionError,
  };
}
```
```
//and here's where the PKPEthers signing failed (because I referenced it earlier in the context of the call to provider.getSessionSigs()
//useAuthenticateAndFetchJWT.tsx
import { useEffect, useState } from 'react';
import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
import ky from 'ky';
import { useLocalStorage } from '@rehooks/local-storage';
import { IRelayPKP, SessionSigs } from '@lit-protocol/types';
import { isJwtExpired } from '../../utils/app';
import { NonceData } from '../../types/types';
import { useAuthOnboardContext } from '@/contexts';

export function useAuthenticateAndFetchJWT(currentAccount: IRelayPKP | null, sessionSigs: SessionSigs | null) {
  const [userJWT, setUserJWT] = useLocalStorage<string | null>("userJWT");
  const [nonce, setNonce] = useState<string | null>("");
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<Error | null>(null);
  const context = useAuthOnboardContext();

  useEffect(() => {
    if (userJWT && isJwtExpired(userJWT)) {
      setUserJWT(null);
    }
  }, [userJWT]);

  useEffect(() => {
    let isMounted = true;

    const authenticateAndFetchJWT = async (currentNonce: string | null) => {
      setIsLoading(true);
      try {
        if (context?.isLitLoggedIn && sessionSigs && currentAccount && (userJWT === null || isJwtExpired(userJWT) || currentNonce === null)) {
          // Fetch new nonce
          let nonceResponse;
          try {
            nonceResponse = await ky('https://supabase-auth.zach-greco.workers.dev/nonce').json<NonceData>();
            setNonce(nonceResponse.nonce);
            currentNonce = nonceResponse.nonce;
          } catch (e) {
            console.error(e);
            throw new Error(`error fetching nonce:`);
          }
          // Use the nonce to sign a message and fetch a new JWT
          let pkpWallet;
          try {
            pkpWallet = new PKPEthersWallet({
              controllerSessionSigs: sessionSigs,
              pkpPubKey: currentAccount.publicKey,
            });
          } catch (e) {
            console.error("new PKPEthersWallet", e);
            throw new Error(`Wallet Constructor: ${e}`);
          }
          try {
            // 401 is still a question of bad sigs I think
            await pkpWallet.init();
          } catch (e) {
            console.error("pkpWallet.init", e);
            throw new Error(`error initializing pkpWallet`);
          }
          let signature;
          try {
            signature = await pkpWallet.signMessage(nonceResponse.nonce);
          } catch (e) {
            console.error("problem signing nonce", e);
            throw new Error('problem signing');
          }
          let jwtResponse;
          try {
            console.log('run jwt request');
            jwtResponse = await ky.post('https://supabase-auth.zach-greco.workers.dev/jwt', {
              json: { ethereumAddress: currentAccount.ethAddress, signature, nonce: nonceResponse.nonce },
            }).json<{ token: string }>();
            setUserJWT(jwtResponse.token);
          } catch (e) {
            throw new Error(`problem with jwt request to worker: ${e}`);
          }
        }
      } catch (e) {
        console.error("Error final catch", e);
        const errorInstance = e instanceof Error ? e : new Error('An unknown error occurred');
        if (isMounted) {
          setError(errorInstance);
        }
      } finally {
        if (isMounted) {
          setIsLoading(false);
        }
      }
    };

    void (async () => {
      await authenticateAndFetchJWT(nonce);
    })();

    return () => {
      isMounted = false;
    };
  }, [currentAccount, sessionSigs, userJWT]);

  return { cachedJWT: userJWT, nonce, isLoading, error };
}
```
</details>
![Screenshot from 2024-05-11 14-20-39](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/e99ac52f-1eef-4f5f-b1bf-4b6b592c3724)

![Screenshot from 2024-05-11 14-21-11](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/d991752b-31a8-4195-b355-a605bc9aca13)

![Screenshot from 2024-05-11 14-22-01](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/88a61420-e9b7-4b5c-a6f2-58888847b954)

**Comment by zach-is-my-name** [comment]
![Screenshot from 2024-05-11 14-17-26](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/00281d26-f48b-4005-9222-811d0d2dac65)
![Screenshot from 2024-05-11 14-17-42](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/9179b7b3-b3dd-47e6-86a4-27b4409e8d94)

**Comment by zach-is-my-name** [comment]
![Screenshot from 2024-05-12 21-30-39](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/2328c54f-f7f6-4601-b406-6a07623f9a0e)
![Screenshot from 2024-05-12 21-29-15](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/38e06cbb-c1c0-41db-9b88-c508548b8ec8)

**Comment by aditya172926** [comment]
> Just tested this (v5): provider.getSessionSigs(..) ~success~
> 
> ```
> export const litNodeClient: LitNodeClient = new LitNodeClient({
>   alertWhenUnauthorized: false,
>   litNetwork,
>   checkNodeAttestation: false,
>   debug: true
> });
> 
> export const litAuthClient: LitAuthClient = new LitAuthClient({
>   debug: true,
>   litRelayConfig: {
>     relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
>     relayUrl: process.env.NEXT_PUBLIC_LIT_RELAY_URL_M
>   },
>   // litNodeClient,
> });
> ```
> 
> But if I include the litNodeClient in the authClient constructor param object it fails with the timeout
> 
> ```
> export const litAuthClient: LitAuthClient = new LitAuthClient({
>   debug: true,
>   litRelayConfig: {
>     relayApiKey: process.env.NEXT_PUBLIC_LIT_RELAY_API_KEY,
>     relayUrl: process.env.NEXT_PUBLIC_LIT_RELAY_URL_M
>   },
>   litNodeClient,
> });
> ```

Hey @zach-is-my-name I am facing this same issue so will tag along here.
Also just wanted to clarify, did removing either litNodeClient or relay URL from `litAuthClient` work? Briefly I was able to generate session Sigs with that fix, but now I am back to the `Timeout error`.

Network: Habanero
Lit Node Client v5.0.0

**Comment by zach-is-my-name** [comment]
@aditya172926 "Also just wanted to clarify, did removing either litNodeClient or relay URL from litAuthClient work?" 
That aligns the relay network rpc's so it won't fail for that reason.  After that is overcome (fixed for unpublished 6.0.1) a variety of network errors result from getSessionSigs (provider.getSessionSigs in my case, which is completely different than litNodeClient.getSessionSigs() in your case). Timeouts, 502:      "Error fetching block from hash", "couldn't get triple". Are some of the errors for those two methods. *Then*: Even if you get lucky and have session sigs returned.  Trying to do something with them: pkpWalletInstance.signMessage() in my case will fail with a timeout.  I'll be honest it's frustrating because my app is completely reliant on Lit from start to finish, and not having a single functioning network for almost a week (Cayenne returns 0 of 2 signatures) is costing me a lot.

**Comment by aditya172926** [comment]
> @aditya172926 "Also just wanted to clarify, did removing either litNodeClient or relay URL from litAuthClient work?" 
> That aligns the relay network rpc's so it won't fail for that reason.  After that is overcome (fixed for unpublished 6.0.1) a variety of network errors result from getSessionSigs (provider.getSessionSigs in my case, which is completely different than litNodeClient.getSessionSigs() in your case). Timeouts, 502:      "Error fetching block from hash", "couldn't get triple". Are some of the errors for those two methods. *Then*: Even if you get lucky and have session sigs returned.  Trying to do something with them: pkpWalletInstance.signMessage() in my case will fail with a timeout.  I'll be honest it's frustrating because my app is completely reliant on Lit from start to finish, and not having a single functioning network for almost a week (Cayenne returns 0 of 2 signatures) is costing me a lot.      


Hmm I see.
Well eventually I will have to get to the message and txn signing with pkpWalletInstance.
But before that, the timeout error for getSessionSigs will need to be unblocked.

So after reading where this current issue is headed, I think at this point it will be better if I create a separate issue here for the Timeout error.

**Comment by joshLong145** [comment]
@DashKash54 
Tagging you so you can take a look.

**Comment by joshLong145** [comment]
@zach-is-my-name 
we have published a new build the first issue regarding the `relayUrl` which is no available: [5.0.1](https://www.npmjs.com/package/@lit-protocol/lit-node-client/v/5.0.1)

**Comment by zach-is-my-name** [comment]
> @zach-is-my-name we have published a new build the first issue regarding the `relayUrl` which is no available: [5.0.1](https://www.npmjs.com/package/@lit-protocol/lit-node-client/v/5.0.1)

Cool. You're the best.  Where's the backend squad?

**Comment by zach-is-my-name** [comment]
Monday 9:30 UTC -5 hours

Habanero returns sessionSigs but timeout on pkpWalletInstance.signMessage("message").  Please advise if I should open a new issue.  
![Screenshot from 2024-05-13 20-27-45](https://github.com/LIT-Protocol/Issues-and-Reports/assets/16063092/38b3a71d-7846-49fe-b0cd-7fef117a7dc2)

```
import { useEffect, useState } from 'react';
import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
import ky from 'ky';
import { useLocalStorage } from '@rehooks/local-storage';
import { IRelayPKP, SessionSigs } from '@lit-protocol/types';
import { isJwtExpired } from '../../utils/app';
import { NonceData } from '../../types/types';
import { useAuthOnboardContext } from '@/contexts';

export function useAuthenticateAndFetchJWT(currentAccount: IRelayPKP | null, sessionSigs: SessionSigs | null) {
  const [userJWT, setUserJWT] = useLocalStorage<string | null>("userJWT");
  const [nonce, setNonce] = useState<string | null>("");
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<Error | null>(null);
  const context = useAuthOnboardContext();

  useEffect(() => {
    if (userJWT && isJwtExpired(userJWT)) {
      setUserJWT(null);
    }
  }, [userJWT]);

  useEffect(() => {
    let isMounted = true;

    const authenticateAndFetchJWT = async (currentNonce: string | null) => {
      setIsLoading(true);
      try {
        if (context?.isLitLoggedIn && sessionSigs && currentAccount && (userJWT === null || isJwtExpired(userJWT) || currentNonce === null)) {
          // Fetch new nonce
          let nonceResponse = await ky('https://supabase-auth.zach-greco.workers.dev/nonce')
            .json<NonceData>()
            .catch(e => {
              console.error("Error fetching nonce", e);
              throw new Error(`Error fetching nonce: ${e}`);
            });

          if (nonceResponse) {
            console.log('setNonce');
            setNonce(nonceResponse.nonce);
            currentNonce = nonceResponse.nonce;
          } else {
            console.log("setNonce failed");
          }

          // Use the nonce to sign a message and fetch a new JWT
          let pkpWallet = new PKPEthersWallet({
            controllerSessionSigs: sessionSigs,
            pkpPubKey: currentAccount.publicKey,
          });

          await pkpWallet?.init().catch(e => console.error("pkpWallet.init", e));

          if (!nonceResponse?.nonce) throw new Error("no response on nonce");

          let signature = await pkpWallet?.signMessage(nonceResponse.nonce).catch(e => console.error("problem signing nonce", e));

          console.log('run jwt request');
          let jwtResponse = await ky.post('https://supabase-auth.zach-greco.workers.dev/jwt', {
            json: { ethereumAddress: currentAccount.ethAddress, signature, nonce: nonceResponse.nonce },
          })
            .json<{ token: string }>()
            .catch(e => {
              console.error('problem with jwt request to worker', e);
              return null; // Return null if an error occurs
            });

          if (jwtResponse && jwtResponse.token) {
            const token = jwtResponse.token;
            setUserJWT(token);
          } else {
            console.error('Invalid JWT response');
            // Handle the case when the JWT response is invalid or missing the token
          }
        }
      } catch (e) {
        console.error("Error final catch", e);
        const errorInstance = e instanceof Error ? e : new Error('An unknown error occurred');
        if (isMounted) {
          setError(errorInstance);
        }
      } finally {
        if (isMounted) {
          setIsLoading(false);
        }
      }
    };

    void (async () => {
      await authenticateAndFetchJWT(nonce);
    })();

    return () => {
      isMounted = false;
    };
  }, [currentAccount, sessionSigs, userJWT]);

  return { cachedJWT: userJWT, nonce, isLoading, error };
}
```

**Comment by spacesailor24** [comment]
@zach-is-my-name No need to create a separate issue, posted your latest to the node team

**Comment by zach-is-my-name** [comment]
> Hey @zach-is-my-name  - I think you have to pass "litNetwork": "habanero" to the new PKPEthersWallet() constructor to make it talk to habanero.  by default, PKPEthersWallet will talk to Cayenne

Very good. Thank you. I was examining the PkpConstructor params late last night until my motherboard fried. I really appreciate the tip and apologize if this thread when a little haywire because I wasn't aware of that. Congrats on the new deploy

**Comment by zach-is-my-name** [comment]
I’ll be trying this as soon as I get my new machine up and running. If a day or two passes and I don’t report back or close please forgive me. But I will circle back here as soon as I try it

**Comment by team-charli** [comment]
> Hey @zach-is-my-name - I think you have to pass "litNetwork": "habanero" to the new PKPEthersWallet() constructor to make it talk to habanero. by default, PKPEthersWallet will talk to Cayenne

From the new v6 Migrations docs it seems to convey this is not necessary:

> // v5
> import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
> 
> const pkpEthersWallet = new PKPEthersWallet({
>   controllerAuthMethods,
>   controllerAuthSig,
>   pkpPubKey,
>   rpc,
>   litNetwork: globalThis.LitCI.network,
> });
> 
> // v6
> import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
> 
> const pkpEthersWallet = new PKPEthersWallet({
>   controllerSessionSigs,
>   litNodeClient,
>   pkpPubKey,
> });

Source: https://developer.litprotocol.com/v3/sdk/migrations/6.0.0#lit-protocolpkp-ethers

Is the new v6 detecting this somehow? If not maybe the network prop should not be optional?

**Comment by zach-is-my-name** [comment]
> Hey @zach-is-my-name - I think you have to pass "litNetwork": "habanero" to the new PKPEthersWallet() constructor to make it talk to habanero. by default, PKPEthersWallet will talk to Cayenne

From the new v6 Migrations docs it seems to convey this is not necessary:

> // v5
> import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
> 
> const pkpEthersWallet = new PKPEthersWallet({
>   controllerAuthMethods,
>   controllerAuthSig,
>   pkpPubKey,
>   rpc,
>   litNetwork: globalThis.LitCI.network,
> });
> 
> // v6
> import { PKPEthersWallet } from '@lit-protocol/pkp-ethers';
> 
> const pkpEthersWallet = new PKPEthersWallet({
>   controllerSessionSigs,
>   litNodeClient,
>   pkpPubKey,
> });

Source: https://developer.litprotocol.com/v3/sdk/migrations/6.0.0#lit-protocolpkp-ethers

Is the new v6 sdk now detecting the network somehow? If not maybe the litNetwork prop should **not** be optional?
